### DISCUSSION TITLE
Real-time collaborative text editing, for pair programming

### DISCUSSION THREAD

**Discussion by skyfaller** [lapce/lapce#1025] [Ideas]
Has anyone thought about how real-time collaborative text editing, like VS Code's Live Share, could work in Lapce? I'm encouraged by the built-in terminal, that seems like it could help match Live Share's terminal-sharing features. If this worked smoothly I think it could be a killer app for Lapce, people really want something a little more user-friendly than tmux for pair programming, and also want out of the Microsoft-controlled ecosystem.

UPDATE: OK, I see the [Remote Development](https://docs.lapce.dev/get-started/remote-development) feature, but it's not clear to me from the docs, is there a good way to have multiple people editing a file at once using this?

**Comment by MinusGix** [comment]
I don't believe there's been much discussion about implementing pair programming features.  
Potentially some parts of the upcoming plugin api would probably make certain parts feasible (ex: sharing edits as they happen), but would probably also need to add plugin API commands for common things (ex: show other person's cursor? be able to view the file they're focusing on? etcetera, though I don't particularly use them so I am uncertain about all that would be desired). Though, I wouldn't expect a pair programming plugin soon.

**Comment by ssbunny** [comment]
Zed？
